src/cnd/cnd.ts

```diff
@@ -329,6 +329,218 @@ export interface SwapProperties {
   [k: string]: any;
 }
 
+/**
+ * The payload returned when fetching one swap on the `/swaps/:id` endpoint
+ */
+export interface SwapResponse extends Entity {
+  properties?: FullSwapProperties;
+}
+
+/**
+ * Element of the array in the payload returned when fetching all swaps on the `/swaps/` endpoint
+ */
+export interface SwapElementResponse extends EmbeddedRepresentationSubEntity {
+  properties?: PartialSwapProperties;
+}
+
+/**
+ * The overall status of a swap
+ */
+export enum SwapStatus {
+  /**
+   * The swap was created but the communication phase hasn't yet been finalized
+   */
+  Created = "CREATED",
+  /**
+   * The communication was finalized and blockchain actions are needed or happening
+   */
+  InProgress = "IN_PROGRESS",
+  /**
+   * The swap is finished and the assets were swapped
+   */
+  Swapped = "SWAPPED",
+  /**
+   * The swap is finished and the assets were not swapped
+   */
+  NotSwapped = "NOT_SWAPPED",
+  /**
+   * An unexpected internal failure aborted the swap
+   */
+  InternalFailure = "INTERNAL_FAILURE"
+}
+
+/**
+ * The parameters for one side of the swap
+ */
+export interface Parameters {
+  // Do we want enums instead?
+  /**
+   * Protocol: "han", "herc20" or "halight"
+   */
+  protocol: string;
+  /**
+   * Ledger: "ethereum", "bitcoin" or "lightning"
+   */
+  ledger: string;
+  /**
+   * Network of the ledger: "mainnet", "testnet", "regtest" for Bitcoin, a chain id number for Ethereum
+   */
+  // This is tricky, an alternative would be to have more nesting where each ledger type (Bitcoin, Ethereum, Lightning) have a specific interface to declare network and asset
+  network: string | number;
+  /**
+   * Asset: "bitcoin", "ether" or { standard: "erc20", contract_address: "0x..." } for non-native tokens
+   */
+  // Not sure this is ideal, let's discuss and see what we can come up with
+  asset: string | { standard: "erc20"; contract_address: "0x..." };
+  /**
+   * Expiry: Expiry applying to this ledger, it can be an absolute or relative expiry, depending on the protocol and ledger
+   */
+  // We could have an object that allows the specification of the type of expiry here
+  expiry: number;
+}
+
+export enum EscrowStatus {
+  None = "NONE",
+  Initialized = "INITIALIZED",
```

Comment:
The distinction between `Initialized` and `Deployed` is too vague for me. I would prefer if we can put some more thought into this before committing to something here.

Reply:
Since this is only for presentational purposes, maybe we can omit some of the nifty details here and decide on a smaller, fixed set of values?

Reply:
I think this set of value makes sense because of another train of thought I had recently. I have added documentation to clarify those.